clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -1063,6 +1063,15 @@ CIRGenModule::getConstantArrayFromStringLiteral(const StringLiteral *E) {
     Str.resize(finalSize);
 
     auto eltTy = getTypes().ConvertType(CAT->getElementType());
+
+    // If the string is full of null bytes, emit a #cir.zero rather than
+    // a #cir.const_array.
+    if (Str.count('\0') == Str.size()) {
+      auto arrayTy =
+          mlir::cir::ArrayType::get(builder.getContext(), eltTy, finalSize);
+      return builder.getZeroAttr(arrayTy);
+    }
+
     return builder.getString(Str, eltTy, finalSize);
```

Comment:
Could you move this logic into the `builder.getString` method? It will facilitate reuse and keep the codegen code closer to its original counterpart.

Reply:
Updated.